src/databases/CGNS/avtCGNSFileReader.C

```diff
@@ -1777,350 +1782,1018 @@ avtCGNSFileReader::GetUnstructuredMesh(int timestate, int base, int zone, const
+                        << "named sections... This shouldn't happen!" << endl;
+                    break;
+                case NFACE_n: 
+                    debug1 << mName << "Found NFACE_n types while processing "
+                        << "named sections... This shouldn't happen!" << endl;
+                    break;
+                default:
                     delete [] elements;
                     elements = 0;
-                    debug4 << mName << cg_get_error() << endl;
-                    continue;
-                }
+                    ugrid->Delete();
+                    EXCEPTION1(InvalidVariableException, meshName);
+                    break;
+            }
+        }
 
-                debug4 << "section " << sec << ": elementType=";
-                PrintElementType(et);
-                debug4 << " start=" << start << " end=" << end << " bound=" << bound
-                       << " interior elements=" << elementSizeInterior
-                       << " parent_flag=" << parent_flag << endl;
+        debug4 << mName << "Done reading cell connectivity." << endl;
+        delete [] elements;
 
-                //
-                // Iterate over the elements and insert them into ugrid.
-                //
-                vtkIdType verts[27];
-                const cgsize_t *elem = elements;
-                for(cgsize_t icell = 0; icell < elementSizeInterior; ++icell)
-                {
-                    // If we're reading mixed elements then the element type
-                    // comes first.
-                    ElementType_t currentType = et;
-                    if(currentType == MIXED)
-                    {
-                        currentType = (ElementType_t)(*elem++);
-#if 0
-                        debug4 << "et[" << icell << "] = ";
-                        PrintElementType(currentType);
-                        debug4 << endl;
+        // Tell the user if we found any higher order elements.
+        if (higherOrderWarning)
+        {
+            avtCallback::IssueWarning("VisIt found quadratic or cubic cells "
+                "in the mesh and reduced them to linear cells. Contact "
+                "visit-users@ornl.gov if you would like VisIt to natively "
+                "process higher order elements.");
+        }
+    }
+}
+
+// ****************************************************************************
+//  Method: avtCGNSFileReader::ReadNGonSections
+//
+//  Purpose:
+//      Read the NGON_n sections, and add them to a given unstructured
+//      grid. The NGON_n sections define point connectivity for faces.
+//      This method assumes that NFACE_n is NOT defined for this mesh
+//      and will add the elements as 2D surface cells.
+//
+//  Arguments:
+//      urgrid                  The unstructured grid to add our cells to.
+//      meshName                The name of our mesh.
+//      mixedAndNamedSections   An array of our mixed and named section indices.
```

Comment:
This should describe nGonSections.

Reply:
Done.

Reply:
Remove mixedAndNamedSections.

Reply:
I don't know how I missed these ones... it should be fixed now.